jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/ClientUpgradeRequest.java

```diff
@@ -161,13 +161,13 @@ public String getHost()
     @Override
     public String getHttpVersion()
     {
-        return httpVersion;
+        throw new UnsupportedOperationException("HttpVersion not available on ClientUpgradeRequest");
```

Comment:
Return the hardcoded value `HttpVersion.HTTP_1_1.asString()`.
We don't want to throw if someone calls this method for logging purposes.

Reply:
What if we're doing websocket over HTTP/2? 
I don't think we have enough information available here to know if we are using HTTP/2 or not. 

The previous implementation of this just returned null for both as there were no setters for it.

Reply:
@lachlan-roberts so why there are these 2 methods then? A relic of 9.4.x?
If we cannot implement them like you correctly point out, let's remove them.

Reply:
We need these methods here because this implements the `org.eclipse.jetty.websocket.api.UpgradeRequest` interface. These methods can be used on the server side, and are implemented on the client after the upgrade once you get it from the `Session`.

Reply:
@lachlan-roberts @sbordet Why is it not possible to specify the protocol of the websocket upgrade in this class, and instead we have to use separate clients? When building a proxy server with jetty, one can't use a single client due to this, because the proxy will not be able to set the http version for web socket proxying(which should be the same as between the source and the proxy).

Reply:
@iiliev2 I don't understand?
Did you actually try and it's not working? 
If so please open a new issue with more details.

Reply:
Is there an official way that I can do this? How can I try it? This class throws an error that http version is not supported.

Reply:
@iiliev2 sorry, but I still don't understand what the problem is.
Why are you calling this method? What use would you do with the return value?
It throws because it cannot return a correct value.

Reply:
I am not calling this method. I was looking for a `setHttpVersion`, then I saw not only is it not available, there is this `get` method which explicitly denies this being possible.

Reply:
Why are you trying to set the HttpVersion? can you explain your use case, are you trying to use WebSocket over HTTP/2?

And the best place for this discussion is a new issue (this PR was merged over 4 years ago).

Reply:
Opened https://github.com/jetty/jetty.project/issues/12659